vms/platformvm/validators/manager.go

```diff
@@ -229,30 +228,22 @@ func (m *manager) makePrimaryNetworkValidatorSet(
 	}
 
 	// Rebuild primary network validators at [height]
-	for diffHeight := currentHeight; diffHeight > targetHeight; diffHeight-- {
-		weightDiffs, err := m.state.GetValidatorWeightDiffs(diffHeight, constants.PlatformChainID)
-		if err != nil {
-			return nil, err
-		}
-		for nodeID, weightDiff := range weightDiffs {
-			if err := applyWeightDiff(validatorSet, nodeID, weightDiff); err != nil {
-				return nil, err
-			}
-		}
-
-		pkDiffs, err := m.state.GetValidatorPublicKeyDiffs(diffHeight)
-		if err != nil {
-			return nil, err
-		}
-		for nodeID, pk := range pkDiffs {
-			if vdr, ok := validatorSet[nodeID]; ok {
-				// The validator's public key was removed at this block, so it
-				// was in the validator set before.
-				vdr.PublicKey = pk
-			}
-		}
+	err := m.state.ApplyValidatorWeightDiffs(
+		validatorSet,
+		currentHeight,
+		targetHeight+1,
```

Comment:
nit: comment on the `targetHeight +1`? IIUC, at height h we store the diffs that changed validators set at height h-1 to height h. So applying diffs backwards till targetHeight+1, rebuild validator state at height targetHeigh.